### ISSUE TITLE
ownCloud not syncing with KOrganizer

### ISSUE THREAD

**Issue by thefirstofthe300** [owncloud/core#23715]
I am having issues with syncing my calendar to KOrganizer.  Currently, I setup the CalDAV link to my ownCloud and KOrganizer finds all of the calendars and address books.  However, the sync does not return anything.  No events are added to my local calendar.  However, if I add an event to the ownCloud calendar in KOrganizer, the event is synced to ownCloud.  I have checked the KDE forums and this seems to be an ownCloud issue.
 
### Steps to reproduce
1. Add ownCloud calendar to KOrganizer.
2. Start sync.
3. No events are added to KOrganizer.

### Expected behaviour
KOrganizer syncs with ownCloud.

### Actual behaviour
KOrganizer doesn't list any events from ownCloud

### Server configuration
**Operating system**: Ubuntu Server 14.04

**Web server:** Fully updated Apache 2

**Database:** 5.5.47-0ubuntu0.14.04.1 (Ubuntu)

**PHP version:** PHP 5.5.9-1ubuntu4.14 (cli) (built: Oct 28 2015 01:34:46)

**ownCloud version:** 9.0

**Updated from an older ownCloud or fresh install:** No

**Where did you install ownCloud from:** Digital Ocean image

**Signing status (ownCloud 9.0 and above):** All good here

**List of activated apps:**

Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.0.0.0
  - dav: 0.1.5
  - documents: 0.12.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_mv: 0.8.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - music: 0.3.10
  - news: 7.1.2
  - notes: 2.0.0
  - notifications: 0.2.3
  - ownbackup: 0.3.8
  - passwords: 17.2
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - tasks: 0.9.0
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - user_external
  - user_ldap

**The content of config/config.php:**
```
{
    "system": {
        "updatechecker": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbtype": "mysql",
        "version": "9.0.0.19",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "vlz",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "instanceid": "ocwvycsyi1ka"
    }
}
```
**Are you using external storage, if yes which one:**

yes; Google Drive

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

**Comment by LukasReschke** [comment]
Can you retry with ownCloud 9.0.1 RC1? https://owncloud.org/install/#testing-development

**Comment by vacy** [comment]
I've the same issue.
I've tried RC1 as advised and it does not change that.

**Comment by thefirstofthe300** [comment]
Just installed RC1 and no dice.  Still having the issue.

**Comment by vacy** [comment]
just tried rc2 too, and still having the issue.
but worth to notice, i've another oc9 installation which was upgraded from 8 and this is working fine there. but my fresh oc9 doesn't.

**Comment by DeathByDenim** [comment]
My bug report ( #23745 ) was marked as a duplicate of this one, so I'll just post the extra info I've found here. The reason why I thought my issue was different from this one is that I can't even get the calendars to show up in KOrganizer in the first place.

I disabled the SSL stuff and used Wireshark to examine the network traffic between akonadi (KOrganizer) and ownCloud. It seems I get a 401 not authorized error when doing some of the PROPFIND for caldav because the authentication header isn't being sent by akonadi. However, it does for some, so maybe that is normal. I've attached the HTTP traffic, maybe something odd strikes any of you?
[traffic.zip](https://github.com/owncloud/core/files/202867/traffic.zip)

I'll try comparing it with ownCloud 8.2.3 to see where the difference could be.

**Comment by ChiefPete** [comment]
I updated my owncloud from 8.2.3 to 9.0 and I'm having sync issues also.

For those that can't see their calendars in Korganizer, the address to access the calendar has changed from .../owncloud/remote.php/caldav/ to .../owncloud/remote.php/dav/  - Notice that it's just dav at the end.

The issue I have with the calendar is events that are created via web browser do not show up in korganizer. The events do show up on my android phone which uses CalDAV-Sync.

Events created in Korganizer do get sync'd to owncloud and then to my phone without issue.

Mark

**Comment by DeepDiver1975** [comment]
Looks like we need to have a closer look on how korganizer works....

FYI: the old caldav endpoint is still valid and should work after upgrade.
In addition we fixed some issues in this area for 9.0.1 ... But you are aware of that already.

Let me see if I can find some KDE somewhere for testing ....

**Comment by DeathByDenim** [comment]
Just to confirm @ChiefPete's findings: Changing the CalDav and CardDav link to ../owncloud/remote.php/dav/ does indeed allow me to properly add the remote calendar and address book to akonadi.

What works:
- New contacts made in KAddressBook show up in the web interface.
- New contacts made in the web interface show up in KAddressBook.
- New appointments made in KOrganizer show up in the web interface.

What doesn't work:
- New appointments made in the web interface do not show up in KOrganizer.

**Comment by vacy** [comment]
About the findings of @ChiefPete:
Is it possible that the defect has been introduced with some backport from 9.0 to 8.2.3?
Because this sync "New appointments made in the web interface to show up in KOrganizer." was working in 8.2.2 and after upgrade to 9.0 too! But doesn't with fresh installation of 9.0.
On the other hand I did not try a fresh 8.2.2 yet ;-)
@ChiefPete was 8.2.3 a fresh installation or already has been updated from some 8.x branch?

**Comment by ChiefPete** [comment]
>@ChiefPete was 8.2.3 a fresh installation or already has been updated from some 8.x branch?

ownCloud had been updated. When I get home, I'll look at my notes to verify the 8.2.x version.

@DeathByDenim ownCloud behavior matches what I see.

Mark

**Comment by DeathByDenim** [comment]
I've updated to 9.0.1 and the old caldav and carddav links now work again. Appointments made in the web interface still don't show up in KOrganizer though.

I did however discover that if you make an appointment in KOrganizer and then change that appointment in the web interface, then those changes will appear in KOrganizer. That is:

1. Create appointment in KOrganizer
2. Sync
3. Change appointment in web interface
4. Sync
5. Changes appear in KOrganizer!


So it's just appointments that are created outside of KOrganizer that are broken somehow.

**Comment by vacy** [comment]
Is it me or is it working since the update to 9.0.1?
I get a part of my appointments synced to akonadi/korganizer.
But some of then not... And I've simply no idea how to troubleshoot that.

**Comment by berberic2** [comment]
9.0.1 nor working here.

**Comment by PVince81** [comment]
Korganizer 5.1.3 works for me (KDE Frameworks 5.21.0) against v9.0.1.

It used to not sync any more and I had to upgrade KDE / Korganizer because it seemed to be a bug on that side.

**Comment by jesusrop** [comment]
I have a similar issue. I have installed owncloud in a ubuntu 16.04server with php7. I am runing korganizer 5.13 in kubuntu 16.06.

When adding my owncloud calendar, some events appear but some do not. Everytime I delete and add the calendar, the same events fail to appear.

I'm getting lots of 401 erroras in the owncloud log

Truth to be told, all events sync OK in both my android and osx devices.

**Comment by thefirstofthe300** [comment]
So I have some new information to show for this issue.  Owncloud is choking on what appears to be a malformed date.

{"reqId":"ZtpAkKgrorhPMBleVwz6","remoteAddr":"XXX.XXX.XXX.XXX","app":"caldav","message":"Exception: {\"Message\":\"The supplied iCalendar datetime value is incorrect: 2016-05-24T00:00:00\",\"Exception\":\"LogicException\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/DateTimeParser.php(188): Sabre\\\\VObject\\\\DateTimeParser::parseDateTime('2016-05-24T00:0...', NULL)\\n#1 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/Property\\\/ICalendar\\\/DateTime.php(172): Sabre\\\\VObject\\\\DateTimeParser::parse('2016-05-24T00:0...', NULL)\\n#2 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/Property\\\/ICalendar\\\/DateTime.php(144): Sabre\\\\VObject\\\\Property\\\\ICalendar\\\\DateTime->getDateTimes(NULL)\\n#3 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/Component\\\/VTodo.php(33): Sabre\\\\VObject\\\\Property\\\\ICalendar\\\\DateTime->getDateTime()\\n#4 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/CalDAV\\\/CalendarQueryValidator.php(291): Sabre\\\\VObject\\\\Component\\\\VTodo->isInTimeRange(Object(DateTime), Object(DateTime))\\n#5 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/CalDAV\\\/CalendarQueryValidator.php(79): Sabre\\\\CalDAV\\\\CalendarQueryValidator->validateTimeRange(Object(Sabre\\\\VObject\\\\Component\\\\VTodo), Object(DateTime), Object(DateTime))\\n#6 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/CalDAV\\\/CalendarQueryValidator.php(41): Sabre\\\\CalDAV\\\\CalendarQueryValidator->validateCompFilters(Object(Sabre\\\\VObject\\\\Component\\\\VCalendar), Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/CalDAV\\\/Backend\\\/AbstractBackend.php(145): Sabre\\\\CalDAV\\\\CalendarQueryValidator->validate(Object(Sabre\\\\VObject\\\\Component\\\\VCalendar), Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/caldav\\\/caldavbackend.php(789): Sabre\\\\CalDAV\\\\Backend\\\\AbstractBackend->validateFilterForObject(Array, Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/CalDAV\\\/Calendar.php(427): OCA\\\\DAV\\\\CalDAV\\\\CalDavBackend->calendarQuery('9', Array)\\n#10 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/CalDAV\\\/Plugin.php(600): Sabre\\\\CalDAV\\\\Calendar->calendarQuery(Array)\\n#11 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/CalDAV\\\/Plugin.php(236): Sabre\\\\CalDAV\\\\Plugin->calendarQueryReport(Object(Sabre\\\\CalDAV\\\\Xml\\\\Request\\\\CalendarQueryReport))\\n#12 [internal function]: Sabre\\\\CalDAV\\\\Plugin->report('{urn:ietf:param...', Object(Sabre\\\\CalDAV\\\\Xml\\\\Request\\\\CalendarQueryReport), 'calendars\\\/danse...')\\n#13 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#14 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(716): Sabre\\\\Event\\\\EventEmitter->emit('report', Array)\\n#15 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpReport(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#16 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#17 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:REPORT', Array)\\n#18 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#19 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/caldav.php(76): Sabre\\\\DAV\\\\Server->exec()\\n#20 \\\/var\\\/www\\\/owncloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/ownclo...')\\n#21 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/DateTimeParser.php\",\"Line\":40,\"User\":\"dansey\"}","level":4,"time":"2016-05-14T05:15:06+00:00","method":"REPORT","url":"\/remote.php\/caldav\/calendars\/dansey\/personal\/","user":"dansey"}

**Comment by dances-with-clouds** [comment]
I can add some information that might help to find the bug: 

I just upgraded from OC 8.x to OC 9 (NextCloud) and I am experiencing exactly the same behaviour as described in the first article of this thread: events created in korganizer are synchronized into NC and thus to all clients, but korganizer does not show old entries. And here is the important additional information I can give: korganizer does show *some* events, but only those events, that have been entered (or synched) into NC since the upgrade!

In other words: korganizer does not show any event, that has been in OC *before* the update, no matter if it's a single event or a repeating event.

**Comment by DeepDiver1975** [comment]
> 2016-05-24T00:00:00

well - this is indeed an invalid date time format - it should be 20160524T00:00:00

Any chance to get this fixed within the databse?

**Comment by ballfire** [comment]
I have arrived to this discussion because my korganizer has stopped syncing my owncloud created events; however, I have noticed that the .ics files created by owncloud lack the VERSION:X.0 part and i think this is why the korganizer is not synchronizing the events, they do not conform to the version check korganizer runs on them.

For those who  really know the vcalendar format... Is it mandatory to include the version tag ? i.e. Is it a korganizer bug because it does not accept versionless files or is it an owncloud bug because it creates .ics files without the proper version ?

**Comment by ballfire** [comment]
In my tests I have developed a workaround I am using; it seems the problem goes all the way to ical.js library

Aparently, the ICAL.Component constructor does not include the version property, which is mandatory for ics files.

This workaround modifies the ICalFactor service to add the missing version property besides the prodid property. With this applied the events created in owncloud sync properly in korganizer

But maybe someone who knows better than me how this bug report works should elevate a bug to ical.js

(Full path is owncloud/apps/calendar/js/public)

--- app_original.js     2016-08-31 10:58:46.115159576 +0200
+++ app.js      2016-08-31 10:53:08.247683455 +0200
@@ -4436,6 +4436,7 @@
 
                        var version = angular.element('#fullcalendar').attr('data-appVersion');
                        root.updatePropertyWithValue('prodid', '-//ownCloud calendar v' + version);
+                       root.addPropertyWithValue('version','2.0');
 
                        return root;
                }

**Comment by DeathByDenim** [comment]
Funnily enough there was an update to the Calendar app just yesterday, which seemed to fix my syncing problems.

On closer examination of the source it seems they implemented the same workaround/fix as @ballfire posted.

**Comment by ballfire** [comment]
In fact, someone reported the bug 14 days ago:

https://github.com/owncloud/calendar/issues/706

**Comment by PVince81** [comment]
Okay, thanks for the feedback. Closing then.
